### DISCUSSION TITLE
InstantSearch: Offer way to filter out facets with no values

### DISCUSSION THREAD

**Discussion by ChristianMattarVT** [algolia/instantsearch#5281] [Ideas]
After setting up which facets to display in which way via "Facet display (renderingContent)" section, Instantsearch will always render the entire list of facets configured there - even those, for which no values are in the result set.

<img width="1323" alt="FacetsWithoutValues" src="https://user-images.githubusercontent.com/102296340/159909397-e5d2bd04-2e13-4e6d-bd5c-c02b36810556.png">

We attempted to build this filtering ourselves by hiding facets with no values. However, this led to an issue when using the "Facet value search" feature: In cases where all filter values would be filtered out by entering an invalid search term, the entire facet would suddenly disappear.

Would appreciate support for this in the InstantSearch framework itself!

**Comment by Haroenv** [comment]
I think based on the screenshot you're using a panel, which means you can use the [hidden](https://www.algolia.com/doc/api-reference/widgets/panel/js/#widget-param-hidden) parameter to conditionally hide the filter.

```js
panel({
  hidden(options) {
    return !options.isFromSearch && options.items.length === 0;
  }
})...
```

If that's not the code you're using to rendering, could you post a sample of what you have so far?

**Comment by ChristianMattarVT** [comment]
Thanks Haroenv! I actually filed this in the wrong component.
Could you please move it to React project? If not possible, I will refile there.
